src/utils/service-utilities/utils.ts

```diff
@@ -400,7 +414,8 @@ export const parseTextforVaultConfig = (secret: string) => {
     const { miniscript, keyInfoMap } = miniscriptScheme;
     let walletPolicyDescriptor = miniscript;
     for (const keyId in keyInfoMap) {
-      walletPolicyDescriptor = walletPolicyDescriptor.replace(keyId, keyInfoMap[keyId]);
+      walletPolicyDescriptor = walletPolicyDescriptor.replace(`(${keyId}`, `(${keyInfoMap[keyId]}`);
+      walletPolicyDescriptor = walletPolicyDescriptor.replace(`,${keyId}`, `,${keyInfoMap[keyId]}`);
```

Comment:
Why do we have the walletPolicyDescriptor being updated twice?

Reply:
The problem was replacing first K1 also replaced IK1 (withot removing the I), so I added a check that it's the beginnign of the expression ie by checking if first there's ( or ,